sumologic/resource_sumologic_muting_schedules_library_muting_schedule.go

```diff
@@ -0,0 +1,324 @@
+
+		"is_system": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Computed: true,
+		},
+
+		"is_mutable": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Computed: true,
+		},
+
+		"created_by": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
+
+		"created_at": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
+
+		"modified_by": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
+
+		"modified_at": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
+	}
+
+	for k, v := range additionalAttributes {
+		tfSchema[k] = v
+	}
+
+	return tfSchema
+}
+
+func resourceSumologicMutingSchedulesLibraryMutingScheduleCreate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	if d.Id() == "" {
+		mutingSchedule := resourceToMutingSchedulesLibraryMutingSchedule(d)
+		if mutingSchedule.ParentID == "" {
+			rootFolder, err := c.GetMutingSchedulesLibraryFolder("root")
+			if err != nil {
+				return err
+			}
+
+			mutingSchedule.ParentID = rootFolder.ID
+		}
+		paramMap := map[string]string{
+			"parentId": mutingSchedule.ParentID,
+		}
+		mutingScheduleDefinitionID, err := c.CreateMutingSchedulesLibraryMutingSchedule(mutingSchedule, paramMap)
+		if err != nil {
+			return err
+		}
+		d.SetId(mutingScheduleDefinitionID)
+	}
+	return resourceSumologicMutingSchedulesLibraryMutingScheduleRead(d, meta)
+}
+
+func resourceSumologicMutingSchedulesLibraryMutingScheduleRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	mutingSchedule, err := c.MutingSchedulesRead(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if mutingSchedule == nil {
+		log.Printf("[WARN] MutingSchedule not found, removing from state: %v - %v", d.Id(), err)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("created_by", mutingSchedule.CreatedBy)
+	d.Set("created_at", mutingSchedule.CreatedAt)
+	d.Set("modified_by", mutingSchedule.ModifiedBy)
+	d.Set("is_mutable", mutingSchedule.IsMutable)
+	d.Set("version", mutingSchedule.Version)
+	d.Set("description", mutingSchedule.Description)
+	d.Set("name", mutingSchedule.Name)
+	d.Set("parent_id", mutingSchedule.ParentID)
+	d.Set("modified_at", mutingSchedule.ModifiedAt)
+	d.Set("content_type", mutingSchedule.ContentType)
+	d.Set("is_system", mutingSchedule.IsSystem)
+	d.Set("monitor", mutingSchedule.Monitor)
+	d.Set("schedule", mutingSchedule.Schedule)
+
+	return nil
+}
+
+func resourceSumologicMutingSchedulesLibraryMutingScheduleUpdate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+	mutingSchedule := resourceToMutingSchedulesLibraryMutingSchedule(d)
+
+	mutingSchedule.Type = "MutingSchedulesLibraryMutingScheduleUpdate"
+	err := c.UpdateMutingSchedulesLibraryMutingSchedule(mutingSchedule)
+	if err != nil {
+		return err
+	}
+	updatedMutingSchedule := resourceSumologicMutingSchedulesLibraryMutingScheduleRead(d, meta)
+	return updatedMutingSchedule
```

Comment:
can we not simply say 
return resourceSumologicMutingSchedulesLibraryMutingScheduleRead(d, meta)

Reply:
I think this is the pattern name conversion for other Contents too. for code consistency, that is the pattern I am currently following.  But welcome any new name rules.

Reply:
This is a minor one. I don't see it is a pattern that is significant that is a practice to be followed.

Reply:
do you have any suggestions? is it OK to keep at it now?